s4e-backend/src/test/java/pl/cyfronet/s4e/data/repository/query/QueryBuilderTest.java

```diff
@@ -62,7 +71,57 @@ public void shouldReturnQueryWithoutAnyConditions() {
         query.append("metadata_content,scene_content,timestamp ");
         query.append("FROM Scene ");
         query.append("WHERE true  ORDER BY id DESC LIMIT ?  OFFSET ? ;");
-        queryBuilder.prepareQueryAndParameters(params, parameters, resultQuery);
+        queryBuilder.prepareQueryAndParameters(params, parameters, resultQuery, errors);
         assertThat(resultQuery.toString(), is(equalTo(query.toString())));
     }
+
+    @Test
+    public void shouldReturnErrorsForTime() {
+        Errors errors = new MapBindingResult(new HashMap<>(), "params");
+        List<Object> parameters = new ArrayList<>();
+        StringBuilder resultQuery = new StringBuilder();
+        Map<String, Object> params = new HashMap<>();
+        params.put(INGESTION_FROM, "2019-11-08");
+        queryBuilder.prepareQueryAndParameters(params, parameters, resultQuery, errors);
+        assertThat(errors.hasErrors(), is(true));
+        assertThat(errors.getFieldError(INGESTION_FROM).getDefaultMessage(),
+                is(equalTo("Text '2019-11-08' could not be parsed at index 10")));
+
+        errors = new MapBindingResult(new HashMap<>(), "params");
+        parameters = new ArrayList<>();
+        resultQuery = new StringBuilder();
+        params = new HashMap<>();
+        params.put(INGESTION_FROM, "Hakuna matata");
+        queryBuilder.prepareQueryAndParameters(params, parameters, resultQuery, errors);
+        assertThat(errors.hasErrors(), is(true));
+        assertThat(errors.getFieldError(INGESTION_FROM).getDefaultMessage(),
+                is(equalTo("Text 'Hakuna matata' could not be parsed at index 0")));
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    public void shouldReturnErrorsForParam(String param, Object value, boolean correct) {
+        Errors errors = new MapBindingResult(new HashMap<>(), "params");
+        List<Object> parameters = new ArrayList<>();
+        StringBuilder resultQuery = new StringBuilder();
+        Map<String, Object> params = new HashMap<>();
+        params.put(param, value);
+        queryBuilder.prepareQueryAndParameters(params, parameters, resultQuery, errors);
+        assertThat(errors.hasErrors(), is(correct));
+    }
+
+    private static Stream<Arguments> shouldReturnErrorsForParam() {
+        return Stream.of(
+                Arguments.of(LIMIT,"abc", true),
+                Arguments.of(LIMIT,"-1", true),
+                Arguments.of(LIMIT,"1000", false),
+                Arguments.of(OFFSET,"abc", true),
+                Arguments.of(OFFSET,"-1", true),
+                Arguments.of(CLOUD_COVER,"abc", true),
+                Arguments.of(CLOUD_COVER,"-1", true),
+                Arguments.of(CLOUD_COVER,"1000", true),
+                Arguments.of(INGESTION_FROM,"2019-11-08", true),
+                Arguments.of(INGESTION_FROM,"Hakuna matata", true)
```

Comment:
You've missed the correct parameters here. I'd put them in the same test.
Also, I'd test more thoroughly the edges of the abstraction classes, e.g. cloud cover = -1, 0, 100, 101, NaN, +inf, -inf.